incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -15,6 +15,27 @@ import (
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 )
 
+func IsSelectorsMatch(inst *unstructured.Unstructured, nsLabels labels.Set) (bool, error) {
+	selector, err := GetSelector(inst)
+	if err != nil {
+		return false, err
+	}
+	treeSelector, err := GetTreeSelector(inst)
+	if err != nil {
+		return false, err
+	}
+	noneSelector, err := GetNoneSelector(inst)
+	if err != nil {
+		return false, err
+	}
+	// We only check notMatch here because an empty selector annotation will be treated as an
+	// "all-match" selector
+	notMatch := (selector == nil || selector.Matches(nsLabels)) &&
+		(treeSelector == nil || treeSelector.Matches(nsLabels)) &&
+		noneSelector == false
+	return notMatch, nil
```

Comment:
This logic isn't very clear to me. Since we're in a function and you can early-return, maybe take advantage of that?

Reply:
Updated to 
```

	if selector != nil && !selector.Matches(nsLabels) {
		return false, nil
	}
	if treeSelector != nil && !treeSelector.Matches(nsLabels) {
		return false, nil
	}
	if noneSelector {
		return false, nil
	}
	return true, nil
```
Does this look better for you?